source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -54,6 +54,46 @@ void SingleServerRequest::onFailure() {
   callbacks_.onResponse(Utility::makeError(Response::get().UpstreamFailure));
 }
 
+bool SingleServerRequest::onMovedRedirection(Common::Redis::RespValue& value) {
+  if ((incoming_request_.get() == nullptr) || (value.type() != Common::Redis::RespType::Error)) {
+    return false;
+  }
+  std::vector<absl::string_view> err = StringUtil::splitToken(value.asString(), " ", false);
+  if (err.size() != 3) {
+    return false;
+  }
+
+  const std::string host_address = std::string(err[2]); // ip:port
+  handle_ = conn_pool_.redirectRequest(host_address, *incoming_request_, *this);
+  return (handle_ != nullptr);
+}
+
+bool SingleServerRequest::onAskRedirection(Common::Redis::RespValue& value) {
+  if ((incoming_request_.get() == nullptr) || (value.type() != Common::Redis::RespType::Error)) {
+    return false;
+  }
+  std::vector<absl::string_view> err = StringUtil::splitToken(value.asString(), " ", false);
+  if (err.size() != 3) {
+    return false;
+  }
+
+  if (incoming_request_->asArray().size() > 0 &&
```

Comment:
Is there any case where we would get an ASK where the arguments after the operation wouldn't work with the ASK that we replace the operation with?

Reply:
It may be cleaner to create a different copy of the object, but it requires more memory and processor cycles to create a full copy. this is a lot of extra work for little gain, as the RespValue is immediately encoded into a buffer at the bottom of the ConnPool::redirectRequest() call.

 I also wanted to handle the case when a request might be redirected multiple times (via ask or moved, or some combination of these). In that case, I don't want to leave a prepended "asking" command on the incoming_request_ RespValue, as it might get accidentally used in another context. It seems much cleaner to prepend "asking", create and encode the new RespValue, and then clean up the original -- so later redirection doesn't cause any problems.

Reply:
I am no longer modifying the original RespValue for SingleServerRequests